test/test_client.py

```diff
@@ -210,10 +210,26 @@ def test_getattr(self):
         self.assertIn("has no attribute '_does_not_exist'", str(context.exception))
 
     def test_iteration(self):
-        def iterate():
-            [a for a in self.client]
-
-        self.assertRaises(TypeError, iterate)
+        client = self.client
+        if "PyPy" in sys.version:
+            msg = "'NoneType' object is not callable"
```

Comment:
I opened https://foss.heptapod.net/pypy/pypy/-/issues/3716 for PyPy to consider improving this error message.